src/function/cast/decimal_cast.cpp

```diff
@@ -114,11 +114,41 @@ struct DecimalScaleDownOperator {
 	}
 };
 
+// This function detects if we can scale a decimal down to another.
+template <class INPUT_TYPE>
+bool CanScaleDownDecimal(INPUT_TYPE input, DecimalScaleInput<INPUT_TYPE> &data) {
+	int64_t divisor = static_cast<int64_t>(pow(10, data.source_scale));
```

Comment:
We should use `NumericHelper::POWERS_OF_TEN` here as well - that would be faster, and avoid potential inaccuracies